pages/api/auth/[...nextauth].js

```diff
@@ -93,6 +93,19 @@ function getCallbacks (req, res) {
      * @return {object}            JSON Web Token that will be saved
      */
     async jwt ({ token, user, account, profile, isNewUser }) {
+      // Check if the user account is deleted
+      if (user?.id) {
+        const dbUser = await prisma.user.findUnique({
+          where: { id: user.id },
+          select: { deletedAt: true }
+        })
+
+        if (dbUser?.deletedAt) {
+          // Return false to prevent sign in for deleted accounts
+          return false
+        }
+      }
```

Comment:
Ah, you're already doing this here in the JWT callback.

But isn't this enough? I think this callback runs on every request to verify the token so if you return `false` (or `null`?), there should be no session.

So I am confused why we need to check if the account is deleted in other places, too.